### ISSUE TITLE
Fix Sxxxx FP/FN: Issue title

### ISSUE THREAD

**Issue by kelvingmoreira** [SonarSource/sonar-dotnet#8110]
### Description

[S4275](https://sonarsource.github.io/rspec/#/rspec/S4275/csharp) raises a false positive when analyzing overridden properties in C# code.

### Repro steps

```
public class A 
{
    private bool _isSelected;

    public virtual bool IsSelected
    {
        get => _isSelected;
        set
        {
            _isSelected = value;
           OnSelectionChanged(value);
        }
    }

    [...]
}

public class B : A
{
    public override IsSelected
    {
        get => base.IsSelected;
        set
        {
            base.IsSelected = value;
            NotifySelectionState();
        }
    }

    [...]
}
```

### Expected behavior

No issue should be raised in this case, since _isSelected is not accessible in the derived class.

### Actual behavior

False positives are raised.

### Known workarounds

pragma disable.

### Related information

* SonarLint version: 7.3.0.77872
* C# version: 8.0

**Comment by antonioaversa** [comment]
Hi @kelvingmoreira, thanks for reporting the issue!

I can reproduce it, and I confirm that this scenario should be treated as a False Positive. 
I have created a [PR to reproduce the False Positive](https://github.com/SonarSource/sonar-dotnet/pull/8138) in our unit tests, and I am adding this issue to our backlog.

The issue is going to be fixed in one of our future hardening sprints.
For the time being:
- If you are using SonarLint with SonarQube or SonarCloud in connected mode: you can mark the issue as a false-positive in the UI. The rule can also be disabled in your Quality Profile if it’s particularly noisy.
- If you are using SonarLint in Standalone Mode: you can use `#pragma warning disable` and `#pragma warning restore` to locally disable the rule in the source, or the `.editorconfig` files to disable or change the severity of the rule.

**Comment by eXpl0it3r** [comment]
Any update on this?

**Comment by simondrum** [comment]
@antonioaversa Hi, PR's merged, but there's no milestone on this issue. An idea on when it will be published?

**Comment by sebastien-marichal** [comment]
Hello @simondrum, @eXpl0it3r ,

The PR merged for this issue is only the reproducer in our UTs.
As of today, we didn't fix this false positive.

In order to plan it soon, I will update the priority on this issue from `Medium` to `High` so it might be picked up during our next hardening effort.

You can consider the fix implemented once this issue will be closed.

Have a nice day!

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1678](https://sonarsource.atlassian.net/browse/NET-1678)

[NET-1678]: https://sonarsource.atlassian.net/browse/NET-1678?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by alex-meseldzija-sonarsource** [comment]
Hi @kelvingmoreira , @eXpl0it3r, @simondrum 

Just informing this has been fixed in our 10.21 analyzer release
